src/Surfnet/StepupMiddleware/ManagementBundle/Controller/InstitutionConfigurationController.php

```diff
@@ -0,0 +1,128 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupMiddleware\ManagementBundle\Controller;
+
+use DateTime;
+use GuzzleHttp;
+use Liip\FunctionalTestBundle\Validator\DataCollectingValidator;
+use Rhumsaa\Uuid\Uuid;
+use Surfnet\StepupMiddleware\ApiBundle\Exception\BadCommandRequestException;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Command\Command;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Configuration\Command\ReconfigureInstitutionConfigurationOptionsCommand;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Exception\ForbiddenException;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Pipeline\Pipeline;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Pipeline\TransactionAwarePipeline;
+use Surfnet\StepupMiddleware\ManagementBundle\Validator\Constraints\ValidReconfigureInstitutionsRequest;
+use Symfony\Bridge\Monolog\Logger;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Validator\ConstraintViolation;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+final class InstitutionConfigurationController extends Controller
+{
+    public function reconfigureAction(Request $request)
+    {
+        $this->denyAccessUnlessGranted(['ROLE_MANAGEMENT']);
+
+        $configuration = GuzzleHttp\json_decode($request->getContent(), true);
+
+        $violations = $this->getValidator()->validate($configuration, new ValidReconfigureInstitutionsRequest());
+        if ($violations->count() > 0) {
+            throw BadCommandRequestException::withViolations(
+                'Invalid reconfigure institutions request',
+                $violations
+            );
+        }
+
+        $commands = [];
+        foreach ($configuration as $institution => $options) {
+            $command                                  = new ReconfigureInstitutionConfigurationOptionsCommand();
+            $command->UUID                            = (string) Uuid::uuid4();
+            $command->institution                     = $institution;
+            $command->useRaLocationsOption            = $options['use_ra_locations'];
+            $command->showRaaContactInformationOption = $options['show_raa_contact_information'];
+
+            $commands[] = $command;
+        }
+
+        if (empty($commands)) {
```

Comment:
This check could be done earlier, and then exit immediately as well (if it is relevant). Would suggest just to log the amount of commands that are going to be executed, not only log when no commands are going to be executed.